src/NUnitFramework/framework/TestParameters.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// TestParameters class holds any named parameters supplied to the test run
+    /// </summary>
+    public class TestParameters
+    {
+        private readonly Dictionary<string, string> _parameters = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Gets the number of test parameters
+        /// </summary>
+        public int Count
+        {
+            get { return _parameters.Count; }
+        }
+
+        /// <summary>
+        /// Gets a collection of the test parameter names
+        /// </summary>
+        public ICollection<string> Names
+        {
+            get { return _parameters.Keys; }
+        }
+
+        /// <summary>
+        /// Indexer is the only public access to the internal dictionary
+        /// </summary>
+        /// <param name="name">Name of the parameter</param>
+        /// <returns>Value of the parameter or null if not present</returns>
+        public string this[string name]
+        {
+            get { return _parameters.ContainsKey(name) ? _parameters[name] : null; }
+        }
+
+        /// <summary>
+        /// Get method is a simple alternative to the indexer
+        /// </summary>
+        /// <param name="name">Name of the paramter</param>
+        /// <returns>Value of the paramter or null if not present</returns>
+        public string Get(string name)
+        {
+            return _parameters.ContainsKey(name) ? _parameters[name] : null;
+        }
+
+        public string Get(string name, string defaultValue)
+        {
+            return _parameters.ContainsKey(name) ? _parameters[name] : defaultValue;
+        }
+        
+        public int Get(string name, int defaultValue)
```

Comment:
I like the conversions, especially for a common type like `int`, so I don't want to see them dropped. I would prefer to see us write them once rather than consumers rewrite them many times.

An alternative might be to provide a number of string extension methods as a separate enhancement so they could write `.Get("name").AsInt(0)`. An `AsDateTime()` would also be useful for converting string DateTime values to real DateTimes since you can use DateTime in `TestCase`. Just a thought :smile: